packages/trie/src/trie/trie.ts

```diff
@@ -46,11 +56,29 @@ export class Trie {
     this._root = this.EMPTY_TRIE_ROOT
     this._deleteFromDB = opts?.deleteFromDB ?? false
 
+    if (opts?.db) {
+      this._persistRoot = opts.persistRoot ?? true
```

Comment:
Why must one pass a `db` if we want to persist root?

Reply:
I think this means that if you pass a db, it will persist unless `persistRoot` is set to false?

Reply:
Well if a custom DB is not passed in it will default `persistRoot` to `false` :thinking:

Reply:
> Why must one pass a `db` if we want to persist root?

Persisting the root just means we right a specific key in the DB with the root value from the `trie` so it can be retrieved if we reinstantiate the trie.  So, if you have an application where you want to keep a trie around across runs of the script, the only way to find the actual root of the `trie` is to set the root hash in the DB with the `__root__` key (or whatever key you wanted to use, but that's what's used here).

Reply:
So, if you don't pass a db in the constructor, there's no underlying data persisted from that trie so no reason to "persist" the trie since you would just be writing it to an in-memory database that gets destroyed when the script ends.

Reply:
Ok I follow your reasoning, since on `copy` we also copy the root, so if we use memory LevelDB then it does not matter if we persist or not